unittest/casadi/spatial.cpp

```diff
@@ -89,4 +90,52 @@ BOOST_AUTO_TEST_CASE(test_quaternion)
   
 }
 
+BOOST_AUTO_TEST_CASE(test_quaternion_normalize)
+{
+  Eigen::Quaternion<double> quat_double(1, 2, 3, 4);
+  quat_double.normalize();
+
+  Eigen::Quaternion<casadi::SX> quat_casadi(1, 2, 3, 4);
+  pinocchio::quaternion::normalize(quat_casadi);
+
+  for (int i = 0; i < 4; ++i) {
+    BOOST_CHECK_CLOSE(double(quat_casadi.coeffs()[i]), quat_double.coeffs()[i], 1e-6);
+  }
+}
+
+
+BOOST_AUTO_TEST_CASE(test_log3_autodiff_degenerative_case)
+{
+  auto quat_casadi = casadi::SX::sym("quat", 4);  // qx, qy, qz, qw
+
+  // eigen quaternion eats scalar first
+  Eigen::Quaternion<casadi::SX> quat(quat_casadi(3), quat_casadi(0),
+                                     quat_casadi(1), quat_casadi(2));
+  auto so3 = pinocchio::quaternion::log3(quat);
+  auto so3_casadi = casadi::SX::sym("so3", 3);
+  pinocchio::casadi::copy(so3, so3_casadi);
+  auto jac = jacobian(so3_casadi, quat_casadi);
+
+  auto so3_func = casadi::Function("log3_func", {quat_casadi}, {so3_casadi}, {"quat_in"}, {"so3_out"});
+  auto jac_func = casadi::Function("jac_func", {quat_casadi}, {jac}, {"quat_in"}, {"jac_out"});
+
+  auto so3_result = so3_func(casadi::DMDict{{"quat_in", {0, 0, 0, 1}}}).at("so3_out");
+  auto jac_result = jac_func(casadi::DMDict{{"quat_in", {0, 0, 0, 1}}}).at("jac_out");
+
+  Eigen::Vector<double, 3> so3_expected(0, 0, 0);
+  Eigen::Matrix<double, 3, 4> jac_expected;
+  jac_expected << 2, 0, 0, 0,
+                  0, 2, 0, 0,
+                  0, 0, 2, 0;
```

Comment:
Comparing with hand derived analytic solution. Not sure if it is too trivial, but this was the failure before this PR. Jacobian value at zero quaternion was [I, 0] instead of [2*I, 0]